com.musala.simple.students.db/src/main/java/com/musala/simple/students/db/MyMongoDatabase.java

```diff
@@ -0,0 +1,258 @@
+		this.studentsCollection = this.getDatabase().getCollection("students");
+		this.studentsCollection.createIndex(index, new IndexOptions().unique(true));
+	}
+
+	/**
+	 * A MongoDb-specific implementation of the {@link Database#addStudent(Student)}
+	 * method. Extracts the properties of the {@link Student} object as an array of
+	 * Fields and then initializes a {@link Document} object to be added to the
+	 * collection (the MongoDb collection only accepts {@link Document} objects. The
+	 * method then iterates over the Fields array and assigns the Student's
+	 * properties to the Document object. The document object is then inserted to
+	 * the collection using a try-catch block in case a document containing the same
+	 * student id is already in the database. The
+	 * {@link MyMongoDatabase#setStudentsCollection} method specifies that the id
+	 * property must be unique.
+	 * 
+	 */
+	@Override
+	public void addStudent(Student student) {
+		Field[] fields = Student.class.getDeclaredFields();
+		Document newStudent = new Document();
+		for (Field field : fields) {
+			String fieldName = field.getName();
+			field.setAccessible(true);
+			try {
+				newStudent.append(fieldName, field.get(student));
+			} catch (IllegalArgumentException | IllegalAccessException e) {
+				logger.error(String.format("An error occured while trying to add student with name %s to the database.",
+						student.getName()));
+				logger.info("Stacktrace:", e);
+			}
+		}
+		try {
+			this.studentsCollection.insertOne(newStudent);
+		} catch (MongoWriteException e) {
+			if (e.getMessage().startsWith("E11000")) {
+				logger.error(String.format(
+						"Student %s with id %d can not be added to the Database. Student with the same id already exists\n",
+						student.getName(), student.getId()));
+			} else {
+				logger.error(String.format("Problem occured while trying to add %s with id %d to the database.\n",
+						student.getName(), student.getId()));
+				logger.error("Stacktrace:", e);
+			}
+		}
+	}
+
+	@Override
+	public void deleteStudentById(int studentId) {
+
+	}
+
+	/**
+	 * A MongoDb-specific implementation of the {@link Database#getStudentById()}
+	 * method. Searches the database for a document with an id property value equal
+	 * to the studentId param. If not found the method throws an exception. Else it
+	 * creates a student object with the required properties and returns it.
+	 * 
+	 * @param studentId
+	 *            the id to find a student in the database
+	 * @return a Student object with id, name, age and grade
+	 */
+	@Override
+	public Student getStudentById(int studentId) throws StudentNotFoundException {
+		Document query = new Document("id", studentId);
+		query = this.studentsCollection.find(query).first();
+
+		if (query == null) {
+			throw new StudentNotFoundException(ErrorMessage.STUDENT_NOT_EXISTS);
+		}
+
+		List list = new ArrayList(query.values());
+		int id = (int) list.get(1);
+		String name = (String) list.get(2);
```

Comment:
Everywhere you cast directly (String), replace with `Objects.toString(list.get(2))`. In such case you make use of Java 8's built-in anti-null-pointer String prevention. :) Read more about `Objects.toString()`.